model_definitions/pima_python_indb_xgboost/model_modules/training.py

```diff
@@ -0,0 +1,106 @@
+from teradataml import (
+    DataFrame,
+    XGBoost,
+    ScaleFit,
+    ScaleTransform,
+)
+
+from aoa import (
+    record_training_stats,
+    aoa_create_context,
+    ModelContext
+)
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import json
+from collections import Counter
+
+
+def traverse_tree(tree, feature_counter):
+    if 'split_' in tree and 'attr_' in tree['split_']:
+        feature_counter[tree['split_']['attr_']] += 1
+    if 'leftChild_' in tree:
+        traverse_tree(tree['leftChild_'], feature_counter)
+    if 'rightChild_' in tree:
+        traverse_tree(tree['rightChild_'], feature_counter)
+
+
+def compute_feature_importance(trees_json):
+    feature_counter = Counter()
+    for tree_json in trees_json:
+        tree = json.loads(tree_json)
+        traverse_tree(tree, feature_counter)
+    total_splits = sum(feature_counter.values())
+    feature_importance = {feature: count / total_splits for feature, count in feature_counter.items()}
+    return feature_importance
+
+
+def plot_feature_importance(fi, img_filename):
+    feat_importances = pd.Series(fi)
+    feat_importances.nlargest(10).plot(kind='barh').set_title('Feature Importance')
+    fig = plt.gcf()
+    fig.savefig(img_filename, dpi=500)
+    plt.clf()
+    
+def train(context: ModelContext, **kwargs):
+    aoa_create_context()
+    
+    feature_names = context.dataset_info.feature_names
+    target_name = context.dataset_info.target_names[0]
+    entity_key = context.dataset_info.entity_key
+
+    # read training dataset from Teradata and convert to pandas
+    train_df = DataFrame.from_query(context.dataset_info.sql)
+
+    print ("Scaling using InDB Functions...")
+    
+    scaler = ScaleFit(
+        data=train_df,
+        target_columns = feature_names,
+        scale_method = context.hyperparams["scale_method"],
+        miss_value = context.hyperparams["miss_value"],
+        global_scale = context.hyperparams["global_scale"].lower() in ['true', '1'],
+        multiplier = context.hyperparams["multiplier"],
+        intercept = context.hyperparams["intercept"]
```

Comment:
Maybe read those hyper parameters in separate lines and either type cast them or type validate them, to avoid mismatching types going into teradataml functions.